Modules/SubModule_Import/xASL_imp_NII2BIDS_RunPerf.m

```diff
@@ -68,82 +71,88 @@ function xASL_imp_NII2BIDS_RunPerf(imPar, bidsPar, studyPar, subjectSessionLabel
+    [jsonLocal, bJsonLocalM0isFile] = xASL_imp_NII2BIDS_Subject_DefineM0Type(...
+        studyPar, bidsPar, jsonLocal, fullfile(inSessionPath,'M0.nii'), fullfile(bidsPar.strPerfusion,[subjectSessionLabel runLabel]));	
 
     %% 5. BIDSify M0	
-    % Check the M0 files only for the first ASL-image run within a session
-    if iRun == 1 % For first run process also M0, for the second run, only reference it
-        for iReversedPE = 1:2 % Check if AP/PA coding is given for M0
-            % Load the M0 JSON if existing
-            if iReversedPE == 1
-                pathM0In = fullfile(inSessionPath,'M0');
-            else
-                pathM0In = fullfile(inSessionPath,'M0PERev');
-            end
-
-            % Load the M0 JSON
-            if xASL_exist([pathM0In '.json'])
-                jsonM0 = spm_jsonread([pathM0In '.json']);
-            else
-                jsonM0 = struct;
-            end
-
-            if iReversedPE == 1
-                if xASL_exist(fullfile(inSessionPath,'M0PERev.nii'))
-                    strPEDirection = '_dir-ap';
-
-                    jsonM0.PhaseEncodingDirection = 'j-';
-                    jsonLocal.PhaseEncodingDirection = 'j-';
-                else
-                    strPEDirection = '';
-                end
-                if bJsonLocalM0isFile
-                    jsonLocal.M0 = [jsonLocal.M0 strPEDirection '_' bidsPar.strM0scan '.nii.gz'];
-                end
-                % Define the path to the respective ASL
-                jsonM0.IntendedFor = [aslOutLabelRelative '_asl.nii.gz'];
-                pathM0Out = fullfile(outSessionPath,bidsPar.strPerfusion,['sub-' subjectSessionLabel strPEDirection '_' bidsPar.strM0scan]);
-            else
-                jsonM0.PhaseEncodingDirection = 'j';
-                jsonM0.IntendedFor = fullfile(bidsPar.strPerfusion,['sub-' subjectSessionLabel '_dir-ap' '_' bidsPar.strM0scan '.nii.gz']);
-                pathM0Out = fullfile(outSessionPath,bidsPar.strFmap,['sub-' subjectSessionLabel '_dir-pa' '_' bidsPar.strM0scan]);
-            end
-
-            % Create the directory for the reversed PE if needed
-            if iReversedPE == 2 && xASL_exist([pathM0In '.nii']) && ~exist(fullfile(outSessionPath,bidsPar.strFmap),'dir')
-                mkdir(fullfile(outSessionPath,bidsPar.strFmap));
-            end
-
-            % If M0, then copy M0 and add ASL path to the IntendedFor
-            if xASL_exist([pathM0In '.nii'])
-                jsonM0 = xASL_bids_BIDSifyM0(jsonM0, jsonLocal, studyPar, pathM0In, pathM0Out, headerASL);
-
-                % Save JSON to new dir
-                jsonM0 = xASL_bids_VendorFieldCheck(jsonM0);
-                jsonM0 = xASL_bids_JsonCheck(jsonM0,'M0');
-                spm_jsonwrite([pathM0Out '.json'],jsonM0);
+    % Check the M0 files 
```

Comment:
Did you change anything here, or only the tab? Pretty difficult to see :)

Reply:
Yeah, I always go to the history for those kind of things. We changed quite a lot. You can find the [history here](xASL_imp_NII2BIDS_RunPerf.m).

Reply:
That would take me too much time, I think. But I could have reviewed each commit separately; though I don't know if this works nicely with the review

Reply:
There are too many commits - I tried to review some commits separately when they fixed things we worked with Michael together on. But the initial commits with restructuring - unfortunately, impossible to check commit by commit. But the structure looks fine. Tests are perfect. And we simply trust Michael enough that he only restructured stuff and knows what he's doing - he's not green anymore :D